File: src/main/java/com/jcloisterzone/figure/DecinskySneznik.java
package com.jcloisterzone.figure;

import com.jcloisterzone.Player;
import com.jcloisterzone.board.Position;
import com.jcloisterzone.board.pointer.FeaturePointer;
import com.jcloisterzone.feature.Field;
import com.jcloisterzone.feature.Structure;
import com.jcloisterzone.figure.RangeFigure;
import com.jcloisterzone.game.capability.trait.FlowersBonusAffected;
import com.jcloisterzone.game.state.GameState;
import com.jcloisterzone.game.state.PlacedTile;
import io.vavr.collection.Stream;
import io.vavr.Tuple2;

public class DecinskySneznik extends Special implements FlowersBonusAffected, RangeFigure {

    private static final long serialVersionUID = 1L;

    public DecinskySneznik(String id, Player player) {
        super(id, player);
    }

    @Override
    public boolean canBeEatenByDragon(GameState state) {
        return false;
    }

    @Override
    public boolean interactingWithOtherMeeples() {
    	return false;
    }
    
    @Override
    public DeploymentCheckResult isDeploymentAllowed(GameState state, FeaturePointer fp, Structure feature) {
        if (!(feature instanceof Field)) {
            return new DeploymentCheckResult("The Decinsky Sneznik must be placed only on a field.");
        }
        return super.isDeploymentAllowed(state, fp, feature);
    }
    
    public Stream<PlacedTile> getRangeTiles(GameState state) {
    	Position position = getPosition(state);
    	return state.getAdjacentTiles2(position).map(Tuple2::_2).append(state.getPlacedTile(position));
    }
}

File: src/main/ts/com/jcloisterzone/figure/DecinskySneznik.ts
import { Stream } from "../../../io/vavr/SeqTypes.js";
import type { Player } from "../Player.js";
import type { FeaturePointer } from "../board/pointer/FeaturePointer.js";
import { Field } from "../feature/Field.js";
import type { Structure } from "../feature/Structure.js";
import type { GameState } from "../game/state/GameState.js";
import type { PlacedTile } from "../game/state/PlacedTile.js";
import { DeploymentCheckResult } from "./DeploymentCheckResult.js";
import { Special } from "./Special.js";

/** Děčínský Sněžník (mini-expansion) — on a field; scores 1 pt per tile in its
 *  range (the deployed tile + its 4 orthogonal neighbours; 5 full). */
import type { UnaffectedByBarn } from "../game/capability/trait/UnaffectedByBarn.js";

import type { FlowersBonusAffected } from "../game/capability/trait/FlowersBonusAffected.js";

export class DecinskySneznik extends Special implements UnaffectedByBarn, FlowersBonusAffected {
  isFlowersBonusAffected(): true {
    return true;
  }

  static readonly simpleName = "DecinskySneznik";
  isUnaffectedByBarn(): true {
    return true;
  }

  constructor(id: string, player: Player) {
    super(id, player);
  }

  override canBeEatenByDragon(_state: GameState): boolean {
    return false;
  }

  override interactingWithOtherMeeples(): boolean {
    return false;
  }

  override isDeploymentAllowed(state: GameState, fp: FeaturePointer, feature: Structure): DeploymentCheckResult {
    if (!(feature instanceof Field)) {
      return new DeploymentCheckResult("The Decinsky Sneznik must be placed only on a field.");
    }
    return super.isDeploymentAllowed(state, fp, feature);
  }

  getRangeTiles(state: GameState): Stream<PlacedTile> {
    const pos = this.getPosition(state)!;
    const tiles: PlacedTile[] = state
      .getAdjacentTiles2(pos)
      .map((t) => t._2)
      .toArray();
    const self = state.getPlacedTile(pos);
    if (self !== null) tiles.push(self);
    return Stream.ofAll(tiles);
  }
}
